pppd/Makefile.am

```diff
@@ -129,7 +129,9 @@ pppd_SOURCES += ipv6cp.c eui64.c
 endif
 
 if WITH_PAM
-pppd_LIBS += -lpam -ldl
```

Comment:
I don't see a pkg-config pam.pc on my system; is it known under a different name?

Reply:
It's from your macro! It's from ax_check_pam.m4.

Reply:
Yes, I did notice the check-in comment said, maybe use pkgtool; but I couldn't find a relatable .pc file ...

Reply:
Aha, now I understand.

For me, I have:
```
$ pkg-config --libs pam                                                                                                                                                                 
-lpam
```

```sys-libs/pam-1.5.1_p20210622``` (1.5.1 with some commits afterwards) installs:
```

/usr/lib64/pkgconfig/pam.pc
/usr/lib64/pkgconfig/pam_misc.pc
/usr/lib64/pkgconfig/pamc.pc
```

I did some digging and it seems to have been [added](https://github.com/linux-pam/linux-pam/commit/b4f0e2e1f7a18169938eedeff8180e0c75473bb5) upstream _after_ 1.5.1 was released, so it's barely in Gentoo, and unlikely to be in any other distributions. Plus, we can't bank on it 😞. Maybe in 10 years 😆?

Reply:
Fair enough, is it possible to add this via ax_check_pam.m4 as a fallback mechanism? Though only if you care enough wanting to add it.

I did spend quite amount of time getting the openssl.m4 just right (a lot of tuning and testing). And in particularly to allow a user to override it. Also, the pkgtool stuff cannot override the case where the user specified --without-pam or --with-pam=DIR. Also thinking someone may want to setup cross-compilation does complicate things ...

Reply:
Not that the output of the --libs for pam yields much more than just -lpam, try glib or some other libs and the linker / cflags stuff gets really crazy